compiler/passes/src/type_checking/check_statements.rs

```diff
@@ -421,6 +419,7 @@ impl<'a, N: Network> StatementVisitor<'a> for TypeChecker<'a, N> {
         // Set the `has_return` flag.
         self.scope_state.has_return = true;
 
+        // TODO - is this useful, or already checked elsewhere?
```

Comment:
I believe the output type of a function is checked to **not** be a nested tuple in [checker.rs](https://github.com/ProvableHQ/leo/blob/26ce60d995c52182868b62e499ae62d53984c7b2/compiler/passes/src/type_checking/checker.rs#L1286).
This is likely redundant.